### PR TITLE
Add Multporn

### PR DESCRIPTION
**Pull request by MikeZeDev**
https://github.com/manga-download/hakuneko/issues/7204

### CODE REVIEW

File: web/src/engine/websites/MultPorn.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList: Manga[] = [];
+        for (let category of categories) {
+            const mangas = await Common.FetchMangasMultiPageCSS.call(this, provider, category + '?page={page}', 'div.view-content table tr td strong a', 0);
```
Comment (ronny1982) [inline web/src/engine/websites/MultPorn.ts:58]: May run requests for each category in parallel. See `Promise.all` or map/reduce pattern

File: web/src/engine/websites/MultPorn.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            mangaList.push(...mangas);
+        }
+        return mangaList;
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const url = new URL(manga.Identifier, this.URI);
+        const settings = await FetchWindowScript<DrupalSettings>(new Request(url), 'Drupal.settings');
+        const chapterList = [];
```
Comment (ronny1982) [inline web/src/engine/websites/MultPorn.ts:67]: This mutable variable is only used inside the `if` block, may be moved into its corresponding scope

File: web/src/engine/websites/MultPorn.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        } else {
+            return Common.FetchChaptersSinglePageCSS.call(this, manga, 'div.view-content table tr td strong.field-content a');
+        }
+
+    }
+
+    private async GetChaptersFromAjaxPage(manga: Manga, page: number, settings: DrupalSettings): Promise<Chapter[]> {
+
+        const view = Object.values(settings.views.ajaxViews).shift();
+        const params = new URLSearchParams();
```
Comment (ronny1982) [inline web/src/engine/websites/MultPorn.ts:84]: Loop seems redundant `const params = new URLSearchParams(view);`

File: web/src/engine/websites/MultPorn.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+                'X-Requested-With': 'XMLHttpRequest',
+                Origin: this.URI.origin,
+                Referer: new URL(manga.Identifier, this.URI).href
+            }
+        }));
+
+        const { data } = results.find(entry => entry.command == 'insert');
+        const dom = new DOMParser().parseFromString(data, 'text/html');
+        const nodes = dom.querySelectorAll<HTMLAnchorElement>('div.view-content table tr td strong.field-content a');
+        return [...nodes]?.map(chapter => new Chapter(this, manga, chapter.pathname, chapter.text.trim())) ?? [];
```
Comment (ronny1982) [inline web/src/engine/websites/MultPorn.ts:105]: Null coalescing operator `[...]?.` is redundant, an empty array can never be `null`

**Review by ronny1982 — APPROVED** [review]

File: web/src/engine/websites/MultPorn.ts [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        }
+    }
+}
+
+type DrupalResult = {
+    command: string,
+    data :string
+}
+
+@Common.MangaCSS(new RegExp(`^{origin}/(${categories.join('|')})/[^/]+`), 'div.breadcrumb span.last')
```
[the remark this answers is not in the document]
Comment (MikeZeDev) [reply web/src/engine/websites/MultPorn.ts:40]: People may paste 

https://multporn.net/video/3d  > videos : not supported

**Comment by MikeZeDev** [comment]
Btw this website is problematic. 

Take a category : i.e https://multporn.net/comics
We use the subcategories as "manga" :  i.e https://multporn.net/comics/adventure_time
Then we have some "chapters" :  i.e https://multporn.net/comics/a_tomboys_request

This chapter got pictures. We could easily say its a "one chapter manga" to be honest.

As you can see, people may paste  i.e https://multporn.net/comics/a_tomboys_request as a "manga", and it wont work as its a "chapter".

We can hardly distinguish between https://multporn.net/comics/adventure_time and https://multporn.net/comics/a_tomboys_request

File: web/src/engine/websites/MultPorn.ts (hunk 3 above)
Comment (MikeZeDev) [reply web/src/engine/websites/MultPorn.ts:84]: got a TS error as `pager_element` is a number if i do that.

File: web/src/engine/websites/MultPorn.ts (hunk 1 above)
Comment (MikeZeDev) [reply web/src/engine/websites/MultPorn.ts:58]: Done. Unless you have any objection with the use of .flat().

**Pull request merged**